src/core_plugins/metrics/public/components/data_config.js

```diff
@@ -0,0 +1,102 @@
+import React from 'react';
+import createTextHandler from './lib/create_text_handler';
+import createSelectHandler from './lib/create_select_handler';
+import Select from 'react-select';
+import { Code } from './code';
+import { Note } from './note';
+import { Info } from './info';
+import { htmlIdGenerator } from '@elastic/eui';
+import FieldSelect from './aggs/field_select';
+
+export function DataConfig({ onChange, model, fields }) {
+  const handleSelectChange = createSelectHandler(onChange);
+  const handleTextChange = createTextHandler(onChange);
+  const htmlId = htmlIdGenerator();
+  const timerangeModes = [
+    { label: 'Entire timerange', value: 'all' },
+    { label: 'Last value', value: 'last' }
+  ];
+  const timerangeModeHelp = (
+    <Note style={{ width: '300px' }}>
+      This setting controls the timespan used for matching documents.{' '}
+      <Code>Entire timerange</Code> will match all the documents selected
+      in the timepicker. <Code>Last value</Code> will match only the documents
+      for the specified interval from the end of the timerange.
+    </Note>
+  );
+
+  return (
+
+    <div className="vis_editor__container">
+      <div className="vis_editor__vis_config-row">
+        <label className="vis_editor__label" htmlFor={htmlId('index_pattern')}>
+          Index pattern
+        </label>
+        <input
+          id={htmlId('index_pattern')}
+          className="vis_editor__input-grows"
+          type="text"
+          onChange={handleTextChange('index_pattern')}
+          value={model.index_pattern}
+        />
+        <label className="vis_editor__label" htmlFor={htmlId('time_field')}>
+          Time Field
+        </label>
+        <div className="vis_editor__row_item">
+          <FieldSelect
+            id={htmlId('time_field')}
+            restrict="date"
+            value={model.time_field}
+            onChange={handleSelectChange('time_field')}
+            indexPattern={model.index_pattern}
+            fields={fields}
+          />
+        </div>
+        {model.type === 'timeseries' ? (
```

Comment:
These conditions could be collapsed using the new built-in [`Fragment`](https://reactjs.org/docs/fragments.html) component:

```javascript
{model.type === 'timeseries' ? (
  <React.Fragment>
    <label>...</label>
    <input ... />
  </React.Fragment>
) : null}
{model.type !== 'timeseries' ? (
  <React.Fragment>
    <label>...</label>
    <div>...</div>
    {model.timerange_mode === 'last' ? (
      <label>...</label>
      <input ... />
    ) : null}
  </React.Fragment>
) : null}
```

Reply:
I didn't realize there was a new `React.Fragment` component... this will make life a whole lot easier. 👍